opensciencegrid/ospool-cm/prometheus.py

```diff
@@ -31,53 +32,61 @@ def cm_resources_info(collector):
     '''
     resources = {}
 
+    # These are the limits we use to determine if a pslot is being
+    # starved. Memory is in MB, Disk in KB.
+    mem_starvation = 2000
+    disk_starvation = 2000000
+
+    # iterate over all resources
     ads = collector.query(ad_type=htcondor.AdTypes.Startd,
                           constraint="!isUndefined(GLIDEIN_ResourceName)",
-                          projection=["GLIDEIN_ResourceName"])
+                          projection=["GLIDEIN_ResourceName", "CPUs", "State"])
     for ad in ads:
         if ad["GLIDEIN_ResourceName"] not in resources:
-            resources[ad["GLIDEIN_ResourceName"]] = {}
-
+            resources[ad["GLIDEIN_ResourceName"]] = {
+                "total_cpus": 0,
+                "claimed_cpus": 0,
+                "idle_retirement_cpus": 0,
+                "idle_memstarvation_cpus": 0,
+                "idle_diskstarvation_cpus": 0,
+                "idle_other_cpus": 0
+            }
+        r = resources[ad["GLIDEIN_ResourceName"]]
+        r["total_cpus"] += int(ad["CPUs"])
+        if ad["State"] != "Unclaimed":
+            r["claimed_cpus"] += int(ad["CPUs"])
+
+    # classify idle CPUs
+    ads = collector.query(ad_type=htcondor.AdTypes.Startd,
+                          constraint="!isUndefined(GLIDEIN_ResourceName) && PartitionableSlot == true && CPUs >= 1",
```

Comment:
We still have a bunch of static slots in the pool, is it OK to exclude them?

Reply:
I think we will revisit those later. Issues like disk starvation could happen, but it is usually caught in the glidein startup validations.

Reply:
OK, that should be a separate PR anyway.